lib/common/xml.c

```diff
@@ -2596,17 +2647,24 @@ int
 write_xml_file(xmlNode * xml_node, const char *filename, gboolean compress)
 {
     FILE *stream = NULL;
+    unsigned int nbytes = 0;
+    int rc = pcmk_rc_ok;
 
     CRM_CHECK(xml_node && filename, return -EINVAL);
     stream = fopen(filename, "w");
     if (stream == NULL) {
         return -errno;
     }
-    return write_xml_stream(xml_node, filename, stream, compress);
+    rc = write_xml_stream(xml_node, filename, stream, compress, &nbytes);
+    if (rc != pcmk_rc_ok) {
+        return pcmk_rc2legacy(rc);
+    }
+    return (int) nbytes;
 }
 
+// Replace a portion of a dynamically allocated string (reallocating memory)
 static char *
-crm_xml_escape_shuffle(char *text, int start, int *length, const char *replace)
+replace_text(char *text, int start, int *length, const char *replace)
```

Comment:
I haven't looked around, but this function really seems like it'd be more broadly useful.

Reply:
There's a separate proposal floating around to use GString instead of our custom reallocating stuff, so I'd leave it alone until that's decided.